### DISCUSSION TITLE
Since document.execCommand is deprecated, what is the plan for future releases?

### DISCUSSION THREAD

**Discussion by FarooqAlaulddin** [codex-team/editor.js#2214] [Q&A] [answered]
As you probably know that `document.execCommand` method has been deprecated and is not recommended for use in modern web development (See [MDN](https://developer.mozilla.org/en-US/docs/Web/API/Document/execCommand#browser_compatibility), [Stackoverflow Thread](https://stackoverflow.com/questions/60581285/execcommand-is-now-obsolete-whats-the-alternative). 

What is your plan for replacement of this function? Do you have any current recommendations for inlinetool developments?

**Comment by gohabereg** [comment, marked as answer]
Hello @FarooqAlaulddin 

`execCommand` is deprecated for quite a while, however there is still no suitable alternative. The same results might be achieved using Range API and [Input Events Level 2 API](https://w3c.github.io/input-events/). However, Range API implementation differs from browser to browser, and Input Events are not widely supported yet.

In the future we would like to have some robust interface provided by Editor.js API for inline tools, however that is really challenging problem as it requires to disabled default browser behaviour and fully simulate it with more modern APIs. So as for now, `execCommand` is a way to go

**Comment by NazmusSayad** [comment]
Another advantage of execCommand is that it can be used at any position within any element without causing too much confusion.
However, other editors like Editor.js or Tiptap tend to create additional divs or containers, which can make it difficult to maintain a reliable editing experience in some rare cases.
